src/Adapter/PlatformServices.Shared/netstandard1.3/Utilities/ns13DeploymentUtilityBase.cs

```diff
@@ -0,0 +1,317 @@
+
+    internal abstract class DeploymentUtilityBase
+    {
+        protected const string TestAssemblyConfigFileExtension = ".config";
+        protected const string NetAppConfigFile = "App.Config";
+
+        /// <summary>
+        /// Prefix for deployment folder to avoid confusions with other folders (like trx attachments).
+        /// </summary>
+        protected const string DeploymentFolderPrefix = "Deploy";
+
+        public DeploymentUtilityBase()
+            : this(new DeploymentItemUtility(new ReflectionUtility()), new AssemblyUtility(), new FileUtility())
+        {
+        }
+
+        public DeploymentUtilityBase(DeploymentItemUtility deploymentItemUtility, AssemblyUtility assemblyUtility, FileUtility fileUtility)
+        {
+            this.DeploymentItemUtility = deploymentItemUtility;
+            this.AssemblyUtility = assemblyUtility;
+            this.FileUtility = fileUtility;
+        }
+
+        protected FileUtility FileUtility { get; set; }
+
+        protected DeploymentItemUtility DeploymentItemUtility { get; set; }
+
+        protected AssemblyUtility AssemblyUtility { get; set; }
+
+        public bool Deploy(IEnumerable<TestCase> tests, string source, IRunContext runContext, ITestExecutionRecorder testExecutionRecorder, TestRunDirectories runDirectories)
+        {
+            IList<DeploymentItem> deploymentItems = this.DeploymentItemUtility.GetDeploymentItems(tests);
+
+            // we just deploy source if there are no deployment items for current source but there are deployment items for other sources
+            return this.Deploy(source, runContext, testExecutionRecorder, deploymentItems, runDirectories);
+        }
+
+        /// <summary>
+        /// Create deployment directories
+        /// </summary>
+        /// <param name="runContext">The run context.</param>
+        /// <returns>TestRunDirectories instance.</returns>
+        public TestRunDirectories CreateDeploymentDirectories(IRunContext runContext)
+        {
+            var resultsDirectory = this.GetTestResultsDirectory(runContext);
+            var rootDeploymentDirectory = this.GetRootDeploymentDirectory(resultsDirectory);
+
+            var result = new TestRunDirectories(rootDeploymentDirectory);
+            var inDirectory = result.InDirectory;
+            var outDirectory = result.OutDirectory;
+            var inMachineDirectory = result.InMachineNameDirectory;
+
+            this.FileUtility.CreateDirectoryIfNotExists(rootDeploymentDirectory);
+            this.FileUtility.CreateDirectoryIfNotExists(inDirectory);
+            this.FileUtility.CreateDirectoryIfNotExists(outDirectory);
+            this.FileUtility.CreateDirectoryIfNotExists(inMachineDirectory);
+
+            return result;
+        }
+
+        public abstract IEnumerable<string> Deploy(IList<DeploymentItem> deploymentItems, string testSource, string deploymentDirectory, string resultsDirectory);
```

Comment:
Add summary tag for all functions describing the purpose of the method, input parameters are output  parameters it takes.

Reply:
done